src/LabelComponent.jsx

```diff
@@ -0,0 +1,158 @@
+import { useState, useEffect } from "react";
+import "./LabelComponent.css";
+
+const LabelComponent = () => {
+  const [inputValue, setInputValue] = useState("");
+  const [labelsData, setLabelsData] = useState([]);
+  const [filteredSuggestions, setFilteredSuggestions] = useState([]);
```

Comment:
Is it necessary to have both states?
Maybe the `suggestions` state is enough?
```suggestion
  const [suggestedLabels, setSuggestedLabels] = useState([]);
```